### DISCUSSION TITLE
Configurable Onboarding PR body

### DISCUSSION THREAD

**Discussion by ps-jay** [renovatebot/renovate#24800] [Suggest an Idea]
### What would you like Renovate to be able to do?

I would like to provide my own message for the onboarding pull request.

In my organisation we have a Slack channel to discuss Renovate, some wiki page documentation, and a repository of shared configuration presets.  I wanted to alter the body of the pull request so that I could provide my own links and explanation of what Renovate does.

### If you have any ideas on how this should be implemented, please tell us here.

It would involve making `prTemplate` configurable as far as I can tell

https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/onboarding/pr/index.ts#L27

### Is this a feature you are interested in implementing yourself?

Maybe - I'm interested to hear feedback on this idea

**Comment by HonkingGoose** [comment]
>  I wanted to alter the body of the pull request so that I could provide my own links and explanation of what Renovate does.

That would only help you when people go looking in the onboarding PR body though. When you make further changes, that original onboarding PR is no longer up to date (unless you manually edit the PR body...).
And Renovate also updates/deprecates terms regularly, say you explained what a `Master Issue` is a while back in your PR, that terms has since been replaced by `Dependency Dashboard`, so you'd have to update the PR body again.

I don't think putting this information in the onboarding PR is the best place. I'd recommend putting this information in a file in your docs folder or on your wiki. 😉 

By the way Renovate supports using a `description` field to explain each option directly in the `renovate.json` config file. That can help to explain some more complex things, like `packageRules` setup:

<details><summary>Click me to see some examples of the description field</summary>

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "ignorePaths": [
    "examples/",
    "packages/docusaurus/src/server/versions/__fixtures__",
    "packages/docusaurus-1.x",
    "packages/docusaurus-init-1.x",
    "packages/docusaurus-migrate/src/__tests__",
    "website-1.x"
  ],
  "rangeStrategy": "replace",
  "dependencyDashboard": true,
  "lockFileMaintenance": {
    "description": "Refresh lock file weekly on mondays",
    "enabled": true
  },
  "stabilityDays": 3,
  "internalChecksFilter": "strict",
  "labels": ["dependencies"],
  "semanticCommits": "enabled",
  "packageRules": [
    {
      "description": "Require approval before proposing any major updates",
      "dependencyDashboardApproval": true,
      "matchUpdateTypes": ["major"]
    },
    {
      "description": "Disable any fixtures or tests missed in our ignorePaths",
      "enabled": false,
      "matchPaths": ["**/__fixtures__/**", "**/__tests__/**"]
    },
    {
      "description": "Reset semantic commit type/scope for all packages",
      "semanticCommitType": "chore",
      "semanticCommitScope": null,
      "matchPackagePatterns": ["*"]
    },
    {
      "description": "Use v2 scope for anything in package/*",
      "semanticCommitScope": "v2",
      "matchPaths": ["packages/"]
    },
    {
      "description": "Group eslint packages together",
      "groupName": "eslint packages",
      "matchPackagePatterns": ["^eslint"]
    },
    {
      "description": "Group webpack packages together",
      "groupName": "webpack packages",
      "matchPackagePatterns": ["^webpack"]
    },
    {
      "description": "Group postcss packages together",
      "groupName": "postcss packages",
      "matchPackagePatterns": ["^postcss"]
    },
    {
      "description": "Group jest packages together",
      "groupName": "jest packages",
      "matchPackagePatterns": ["jest"]
    },
    {
      "description": "Group react packages together",
      "groupName": "react packages",
      "matchPackagePatterns": ["react"]
    }
  ]
}

```

</details>

Read the [Renovate docs, configuration option: description](https://docs.renovatebot.com/configuration-options/#description)

**Comment by rarkins** [comment]
I don't object to this idea though, even if @HonkingGoose's comments are also valid. We'd want to decide whether to go ahead with this based on how bloated it might make our configuration and/or internal logic. If it's fairly clean then we could go ahead.

**Comment by im-aIex** [comment]
I feel like having a `onboardingPrBodyTemplate` similar to `prBodyTemplate` would be nice, or at least including the `prBodyNotes` in the onboarding PR to allow some level of customization would suffice for what I was looking for.
